### PR TITLE
Associations

### PR DESCRIPTION
**Pull request by akucintavalent**
In this pull request I followed the following instructions:
- Create a class `Classroom` with the following:
  - `@label` instance variable, should be initialized in the constructor.
  - Setter and getter for `@label` (remember about `attr_accessor`).
- Create the `has-many`/`belongs-to` relationship between `Classroom` and `Student`. The following should be implemented:
  - Create the `has-many` side (a classroom has many students).
  - Create the `belongs-to` side (a student belongs to a classroom).
  - Make sure that when adding a student to a classroom it also sets the classroom for the student.
  - Make sure that when setting the classroom for a student it also adds it to the classrooms' students.
- Create a class `Book` with the following:
  - `@title` and `@author` instance variables, should be initialized in the constructor.
  - Setters and getters for instance variables (remember about `attr_accessor`).
- Create a class `Rental` with the following:
  - `@date` instance variable, should be initialized in the constructor.
  - Setter and getter for `@date` (remember about `attr_accessor`).
- Create the `many-to-many` (also `has-many-through`) relationship between `Person` and `Book` using the intermediate class `Rental`. The following should be implemented:
  - Create the `has-many` side of `Book` and `Rental` (a book has many rentals).
  - Create the `belongs-to` side of `Rental` and `Book` (a rental belongs to a book).
  - Create the `has-many` side of `Person` and `Rental` (a person has many rentals).
  - Create the `belongs-to` side of `Rental` and `Person` (a rental belongs to a person).
  - Modify the constructor of `Rental` so `Book` and `Person` are set in it.

### CODE REVIEW

File: classroom.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+class Classroom
+  attr_accessor :label
+
```
Comment (MahmoudBakr23) [inline classroom.rb:4]: - [ ] Please make sure that the `Classroom` class is authorized to read `students` data by adding `attr_reader :students`

File: rental.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+class Rental
+  attr_accessor :date
+
```
Comment (MahmoudBakr23) [inline rental.rb:4]: - [ ] Please make sure that the `Rental` class is authorized to read the `person` and the `book` data in order for the renting process to succeed by adding the `attr_reader :person, :book`

File: student.rb [hunk 3]
```diff
@@ -1,9 +1,12 @@
...
 
 class Student < Person
+  attr_writer :classroom
+
   def initialize(age, name, parent_permission, classroom)
     super(age, name, parent_permission)
-    @classroom = classroom
+    @classroom = classroom # not needed but better safe than sorry
+    classroom.add_student(self)
   end
```
Comment (MahmoudBakr23) [inline student.rb:11]: - [ ] Kindly make sure to add a method for adding rentals to a person (student)

```
  def add_rental(book, date)
    # *The person itself should be sent as a parameter to create the new rental achieving the 'has-many' association
    Rental.new(date, book, self)
  end
```

**Review by MahmoudBakr23 — CHANGES_REQUESTED** [review]
Hello @akucintavalent 😃, 

Great work and effort you put into this project so far.

## Highlights 🚀 

There are some excellent points that you have made perfectly. Keep up! 💪
- Linters are correct
- Professional PR
- Great job so far setting up the associations between the models

However, there is always room for improvement.

> Keep in mind that your long hours of work are appreciated. However, the code review process is for improving the code you wrote, and not for criticizing it. 😄

 ## Required Changes ♻️

-  Please fix the broken MIT link in the README file for a completely professional repo

_Please check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

N/A

Cheers and Happy coding!👏👨‍💻💻

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: classroom.rb (hunk 1 above)
Comment (akucintavalent) [reply classroom.rb:4]: Done

File: rental.rb (hunk 2 above)
Comment (akucintavalent) [reply rental.rb:4]: Done

File: student.rb (hunk 3 above)
Comment (akucintavalent) [reply student.rb:11]: Done

**Comment by AshakaE** [comment]
### Hi @akucintavalent   🙋‍♀️, 

# STATUS APPROVED ✔
   
Your project is complete!  There is nothing else to say other than... it's time to merge it 🔗 🚢 
Congratulations! 🎉 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers 🥂  and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**